### ISSUE TITLE
shared_ptr and stack recursion on destructor calls

### ISSUE THREAD

**Issue by JonathanSalwan** [JonathanSalwan/Triton#753]

**Comment by illera88** [comment]
Any new on this?

**Comment by 0xAcid** [comment]
Apéro

**Comment by JonathanSalwan** [comment]
With: `ulimit -s 1024`

```python
from triton import *

ctx = TritonContext(ARCH.X86_64)
for i in range(10000):
    ctx.processing(Instruction(b"\x48\xff\xc0")) # inc rax
ctx.processing(Instruction(b"\x48\xc7\xc0\x00\x00\x00\x00")) # mov rax, 0
```

**Comment by JonathanSalwan** [comment]
The commit 69ed47d implies a slowdown of the execution. The commit e185197 fixes this slowdown.

Below the time of the unittest `src/testers/unittests/test_simulation.py` for several commits. As we can see, the commit 69ed47d increases the time of the execution and the commit e185197 restores the time of the execution but still keeps the role of the gc in order to fix the isse #753.

```
Mar 30, 2020 e1851972b8c0c6dc35607c60d4102b8b449db2da 34.640s (Fix #753: Only keep shared_ptr at deep step level)
Mar 29, 2020 09d045070bce3af2b1b8bfbcc07f588d053dfeed 54.742s (missing const)
Mar 27, 2020 69ed47d44386c97447860f214718aaa70a296324 58.377s (Fix #753: shared_ptr and stack recursion on destructor calls (missing…)
Mar 26, 2020 4a41acec4744bad5aa3c7d2f3f013e73eb592ec8 36.043s (Fix #865: Instruction(address, opcode, size))
Mar 17, 2020 93f28b0c40164ba33f60847cf33e5d89d0f4c18c 37.296s (Add a new examples about synthetizing obfuscated expressions)
Feb 27, 2020 30cbace692cc8bc42eea9d026b74e41a1977eb5d 37.844s (Merge branch 'dev-v0.8' of ...)
Feb 08, 2020 fb3241e94a3e1d0be9831bfc7a865246ee4c9e30 37.600s (Fix Appveyor)
Dec 05, 2019 8f08fe61a73de30cdca0573d25a292357f480df6 34.217s (Fix include)
Oct 27, 2019 8cc318a7d13d921632d6acf43713fcd90afd4037 36.223s (Add missing initParents (#821))
Oct 26, 2019 809f3afa632376faf9731ae257ec9e468e970073 37.725s (Fixing unittest due to a small stack size on windows)
Oct 24, 2019 e6122cc5fadef602fdda9ec99c4581a439c796b2 36.311s (Use WearkPtr)
Oct 24, 2019 714d63aede9aa6beb24a16479ef76c0b4105ac4a 36.563s (Remove fixme)
```
